### ISSUE TITLE
`Get-WinGetPackage`'s `-Exact` option doesn't function

### ISSUE THREAD

**Issue by wmmc88** [microsoft/winget-cli#4561]
### Brief description of your issue

I'm getting different behavior between `winget list -Exact` and `Get-WinGetPackage -Exact`, and it looks like the `-Exact` option of `Get-WinGetPackage` is bugged. When `-Exact` is passed to `Get-WinGetPackage`, it is always returning nothing.

### Steps to reproduce

```
🪟 Windows 11 Enterprise(10.0.22631) (pwsh) ~
at 08:12:16 PM ❯ winget list --id Microsoft.WindowsTerminal
Name                     Id                                Version      Source
-------------------------------------------------------------------------------
Windows Terminal         Microsoft.WindowsTerminal         1.20.11381.0 winget
Windows Terminal Preview Microsoft.WindowsTerminal.Preview 1.21.1382.0  winget

🪟 Windows 11 Enterprise(10.0.22631) (pwsh) ~
at 08:13:01 PM ❯ winget list --id Microsoft.WindowsTerminal --exact
Name             Id                        Version      Source
---------------------------------------------------------------
Windows Terminal Microsoft.WindowsTerminal 1.20.11381.0 winget

🪟 Windows 11 Enterprise(10.0.22631) (pwsh) ~
at 08:13:03 PM ❯ Get-WinGetPackage -Id Microsoft.WindowsTerminal -Source winget

Name                     Id                                Version      Available Source
----                     --                                -------      --------- ------
Windows Terminal         Microsoft.WindowsTerminal         1.20.11381.0           winget
Windows Terminal Preview Microsoft.WindowsTerminal.Preview 1.21.1382.0            winget


🪟 Windows 11 Enterprise(10.0.22631) (pwsh) ~
at 08:13:20 PM ❯ Get-WinGetPackage -Id Microsoft.WindowsTerminal -Source winget -Exact
```

### Expected behavior

I expected the output of `Get-WinGetPackage -Id Microsoft.WindowsTerminal -Source winget -Exact` to match `winget list --id Microsoft.WindowsTerminal --exact`

### Actual behavior

`Get-WinGetPackage -Id Microsoft.WindowsTerminal -Source winget -Exact` returns nothing

### Environment

```shell
winget info:

Windows Package Manager v1.7.11261
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22631.3737
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.22.11261.0

Winget Directories
-----------------------------------------------------------------------------------------------------------------------
Logs                               %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\Diag…
User Settings                      %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\sett…
Portable Links Directory (User)    %LOCALAPPDATA%\Microsoft\WinGet\Links
Portable Links Directory (Machine) C:\Program Files\WinGet\Links
Portable Package Root (User)       %LOCALAPPDATA%\Microsoft\WinGet\Packages
Portable Package Root              C:\Program Files\WinGet\Packages
Portable Package Root (x86)        C:\Program Files (x86)\WinGet\Packages
Installer Downloads                %USERPROFILE%\Downloads

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale

Admin Setting                             State
--------------------------------------------------
LocalManifestFiles                        Disabled
BypassCertificatePinningForMicrosoftStore Disabled
InstallerHashOverride                     Disabled
LocalArchiveMalwareScanOverride           Disabled
```

winget powershell module info:
```
Get-Module Microsoft.WinGet.Client

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Binary     1.8.1133              Microsoft.WinGet.Client             {Add-WinGetSource, Assert-WinGetPackageManager, D…
```
```

**Comment by wmmc88** [comment]
It seems like the version of `Microsoft.Winget.Client` [available in this repo](https://github.com/microsoft/winget-cli/tree/master/tools/PowerShell/Microsoft.WinGet.Client), is not the same as what is[ published to `PSGallery`](https://www.powershellgallery.com/packages/Microsoft.WinGet.Client/1.8.1133).

**Comment by mdanish-kh** [comment]
Get-WinGetPackage doesn't support an `-Exact` option. Try using `-MatchOption Equals` instead

```
 ~ Get-Help Get-WinGetPackage

NAME
    Get-WinGetPackage

SYNTAX
    Get-WinGetPackage [[-Query] <string[]>] [-Tag <string>] [-Command <string>] [-Count <uint>] [-Id <string>] [-Name <string>] [-Moniker <string>]
    [-Source <string>] [-MatchOption {Equals | EqualsCaseInsensitive | StartsWithCaseInsensitive | ContainsCaseInsensitive}] [<CommonParameters>]


ALIASES
    None


REMARKS
    None
```

As pointed out by the GitHub bot, this is related to:
- #3809

[comment]: <[Policy] PowerShell>
[comment]: <[Policy] Issue-Docs>

**Comment by wmmc88** [comment]
> It seems like the version of `Microsoft.Winget.Client` [available in this repo](https://github.com/microsoft/winget-cli/tree/master/tools/PowerShell/Microsoft.WinGet.Client), is not the same as what is[ published to `PSGallery`](https://www.powershellgallery.com/packages/Microsoft.WinGet.Client/1.8.1133).

Is the updated source available somewhere?

**Comment by mklement0** [comment]
https://github.com/microsoft/winget-cli/tree/master/tools/PowerShell/Microsoft.WinGet.Client appears to be an _obsolete_ [Crescendo](https://learn.microsoft.com/en-us/powershell/utility-modules/crescendo/overview?view=ps-modules)-based implementation that uses `winget.exe` behind the scenes.

The source files for what is published to the PowerShell Gallery is at https://github.com/microsoft/winget-cli/tree/master/src/PowerShell

Indeed, `Get-WinGetPackage` has _no_ `-Exact` switch, as @mdanish-kh has pointed out.

The - somewhat unfortunate - reason that it is accepted nonetheless is that it becomes a _positional argument_ to the `-Query` parameter, which is declared with the `ValueFromRemainingArguments` property:

https://github.com/microsoft/winget-cli/blob/ac468944f761ba1870f6664a4fc28ef5b5a5fde7/src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/Common/FinderCmdlet.cs#L54-L59

In other words - leaving aside whether it makes sense to syntactically allow combining `-Query` with `-Id` - 

```
Get-WinGetPackage -Id Microsoft.WindowsTerminal -Source winget -Exact
```

is the same as:

```
Get-WinGetPackage -Id Microsoft.WindowsTerminal -Source winget -Query '-Exact'
```
